### PR TITLE
Chore: Update @testing-library/user-event to 14.4.3

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

- Updates @testing-library/user-event to 14.4.3
- Update failing tests

### Why is it needed?

- Keep testing infrastructure up-to-date

### CODE REVIEW

**Comment by gu-stav** [comment]
I believe the next run would be successful 🤞🏼 . @markkaylor @remidej also pinging you because I had to update quite a few marketplace tests.

File: packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js [hunk 1]
```diff
@@ -188,7 +189,7 @@ describe('Marketplace page - layout', () => {
     await waitForReload();
 
     const providersTab = screen.getByRole('tab', { name: /providers/i });
-    userEvent.click(providersTab);
+    fireEvent.click(providersTab);
```
Comment (markkaylor) [inline packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js:192]: you can't use `userEvent`?

File: packages/core/admin/admin/src/components/LanguageProvider/tests/index.test.js [hunk 2]
```diff
@@ -54,9 +53,7 @@ describe('LanguageProvider', () => {
 
     expect(screen.getByText('English')).toBeInTheDocument();
 
-    act(() => {
-      userEvent.click(screen.getByText('CHANGE'));
-    });
+    fireEvent.click(screen.getByText('CHANGE'));
```
Comment (markkaylor) [inline packages/core/admin/admin/src/components/LanguageProvider/tests/index.test.js:56]: Could you use `userEvent`?

File: packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/components/WebhookForm/tests/index.test.js [hunk 3]
```diff
@@ -70,11 +69,11 @@ describe('Create Webhook', () => {
...
     render(App);
 
-    userEvent.type(screen.getByLabelText(/name/i), 'My webhook');
-    userEvent.type(screen.getByLabelText(/url/i), 'https://google.fr');
+    fireEvent.change(screen.getByLabelText(/name/i), { target: { value: 'My webhook' } });
+    fireEvent.change(screen.getByLabelText(/url/i), { target: { value: 'https://google.fr' } });
     fireEvent.click(screen.getByRole('checkbox', { name: /entry.create/i }));
 
-    userEvent.click(screen.getByRole('button', { name: /Save/i }));
+    fireEvent.click(screen.getByRole('button', { name: /Save/i }));
```
Comment (markkaylor) [inline packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/components/WebhookForm/tests/index.test.js:76]: sorry same comment 🤦‍♂️ ? can you not use userEvent? I know that kent c dodds recommends it in his article https://kentcdodds.com/blog/common-mistakes-with-react-testing-library

File: packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js (hunk 1 above)
Comment (gu-stav) [reply packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js:192]: You can, but it was easier to use `fireEvent`:

1. it is sufficient in this case I think
2. it doesn't require the test to be async

If you want I can revert back ...

File: packages/core/admin/admin/src/components/LanguageProvider/tests/index.test.js (hunk 2 above)
Comment (gu-stav) [reply packages/core/admin/admin/src/components/LanguageProvider/tests/index.test.js:56]: Let's keep one discussion about this topic open.

File: packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/components/WebhookForm/tests/index.test.js (hunk 3 above)
Comment (gu-stav) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/components/WebhookForm/tests/index.test.js:76]: Let's keep one discussion about that open.

File: packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js (hunk 1 above)
Comment (markkaylor) [reply packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js:192]: Nah that's fine just was curious why. It does seem a bit overkill to make the test async for the one click. Up to you

**Review by markkaylor — APPROVED** [review]

**Pull request merged**